Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -337,41 +432,297 @@ struct HfTrackIndexSkimsCreator {
+  array<array<array<double, 3>, 2>, n3ProngDecays> arrMass3Prong;
+
+  // arrays of 2-prong and 3-prong cuts
+  std::array<LabeledArray<double>, n2ProngDecays> cut2Prong;
+  std::array<std::vector<double>, n2ProngDecays> pTBins2Prong;
+  std::array<LabeledArray<double>, n3ProngDecays> cut3Prong;
+  std::array<std::vector<double>, n3ProngDecays> pTBins3Prong;
+
+  void init(InitContext const&)
+  {
+    arrMass2Prong[hf_cand_prong2::DecayType::D0ToPiK] = array{array{massPi, massK},
+                                                              array{massK, massPi}};
+
+    arrMass2Prong[hf_cand_prong2::DecayType::JpsiToEE] = array{array{massElectron, massElectron},
+                                                               array{massElectron, massElectron}};
+
+    arrMass3Prong[hf_cand_prong3::DecayType::DPlusToPiKPi] = array{array{massPi, massK, massPi},
+                                                                   array{massPi, massK, massPi}};
+
+    arrMass3Prong[hf_cand_prong3::DecayType::LcToPKPi] = array{array{massProton, massK, massPi},
+                                                               array{massPi, massK, massProton}};
+
+    arrMass3Prong[hf_cand_prong3::DecayType::DsToPiKK] = array{array{massK, massK, massPi},
+                                                               array{massPi, massK, massK}};
+
+    arrMass3Prong[hf_cand_prong3::DecayType::XicToPKPi] = array{array{massProton, massK, massPi},
+                                                                array{massPi, massK, massProton}};
+
+    // cuts for 2-prong decays retrieved by json. the order must be then one in hf_cand_prong2::DecayType
+    cut2Prong = {cutsD0ToPiK, cutsJpsiToEE};
+    pTBins2Prong = {pTBinsD0ToPiK, pTBinsJpsiToEE};
+    // cuts for 3-prong decays retrieved by json. the order must be then one in hf_cand_prong3::DecayType
+    cut3Prong = {cutsDPlusToPiKPi, cutsLcToPKPi, cutsDsToPiKK, cutsXicToPKPi};
+    pTBins3Prong = {pTBinsDPlusToPiKPi, pTBinsLcToPKPi, pTBinsDsToPiKK, pTBinsXicToPKPi};
+  }
+
+  /// Method to perform selections for 2-prong candidates before vertex reconstruction
+  /// \param hfTracks is the array of 2 tracks
+  /// \param massHypos is a 2D array containing the mass hypotheses for the 2-prong channels
+  /// \param cutStatus is a 2D array with outcome of each selection (filled only in debug mode)
+  /// \param isSelected ia s bitmap with selection outcome
+  template <typename T1, typename T2, typename T3>
+  void is2ProngPreselected(const T1& hfTracks, T2& cutStatus, T3& whichHypo, int& isSelected)
+  {
+    auto arrMom = array{
+      array{hfTracks[0].px(), hfTracks[0].py(), hfTracks[0].pz()},
+      array{hfTracks[1].px(), hfTracks[1].py(), hfTracks[1].pz()}};
+
+    for (int n2 = 0; n2 < n2ProngDecays; n2++) {
+
+      auto pVec = RecoDecay::PVec(arrMom[0], arrMom[1]);
+      auto pT = RecoDecay::Pt(pVec) - pTTolerance; // add tolerance because of no reco decay vertex
```

Comment:
I missed that it was possible to directly pass the two arrays, fixed

Reply:
Indeed I reverted this because it was not properly working, so I just moved it outside the for loop

Reply:
Which part was not working?

Reply:
I tested it against the version in the `dev` and I got different results even if the selections were the same. I now re-tested adding a print to compare the pT values obtained with the two methods and indeed they are the same. I also checked and I get the same results with the old version of the code in case of same selections, so it is likely that I had a mismatch in the two versions of the `JSON` files in the previous test.

Reply:
Has this been clarified and tested?

Reply:
Hi @vkucera, yes and indeed it works as you suggested (and how it is now implemented). Locally I added a `LOGF` that printed the pT computed with the two methods and they were identical.